app/controllers/resources_controller.rb

```diff
@@ -28,35 +29,80 @@ def create
            status: :created
   end
 
+  # PUT /resource/:id
+  def update
+    authorize! :resource
+
+    begin
+      request_dro = cocina_model(params.except(:action, :controller, :resource, :id).to_unsafe_h)
+    rescue BlobError => e
+      # Returning 500 because not clear whose fault it is.
+      return render build_error('500', e, 'Error matching uploading files to file parameters.')
+    end
+
+    result = BackgroundJobResult.create(output: {})
+    UpdateJob.perform_later(model_params: JSON.parse(request_dro.to_json), # Needs to be sidekiq friendly serialization
+                            signed_ids: signed_ids(params),
+                            background_job_result: result)
+
+    render json: { jobId: result.id },
+           location: result,
+           status: :created
+  end
+
   private
 
   def cocina_model(model_params)
     new_model_params = model_params.deep_dup
-    new_model_params[:version] = 1
     decorate_file_sets(new_model_params)
+    Cocina::Models::DRO.new(new_model_params)
+  end
+
+  def cocina_request_model(model_params)
+    new_model_params = model_params.deep_dup
+    new_model_params[:version] = 1
+    decorate_request_file_sets(new_model_params)
 
     Cocina::Models::RequestDRO.new(new_model_params)
   end
 
   # Decorates the provided FileSets with the information we have in the ActiveStorage table.
   # externalIdentifier is also removed from the request.
-  # rubocop:disable Metrics/AbcSize
   def decorate_file_sets(model_params)
     file_sets(model_params).each do |fileset|
-      fileset[:version] = 1
+      fileset[:version] = model_params[:version]
       fileset.dig(:structural, :contains).each do |file|
-        blob = blob_for_signed_id(file.delete(:externalIdentifier), file[:filename])
-        file[:version] = 1
-        file[:size] = blob.byte_size
-        file[:hasMimeType] = blob.content_type || 'application/octet-stream'
-        declared_md5 = file[:hasMessageDigests].find { |digest| digest.fetch(:type) == 'md5' }.fetch(:digest)
-        calculated_md5 = base64_to_hexdigest(blob.checksum)
-        raise BlobError, "MD5 mismatch for #{file[:filename]}" if declared_md5 != calculated_md5
+        decorate_file(file: file,
+                      external_id: "#{model_params[:externalIdentifier]}/#{file[:filename]}",
+                      version: model_params[:version])
       end
     end
   end
+
+  # rubocop:disable Metrics/AbcSize
+  def decorate_file(file:, version:, external_id: nil)
+    blob = blob_for_signed_id(file.delete(:externalIdentifier), file[:filename])
+    file[:externalIdentifier] = external_id if external_id
+    file[:version] = version
```

Comment:
File version changes even if file hasn't?

Reply:
Again, this is synched to the parent, since we don't track file versions separately in Fedora. I think this could change when we move to a different data store.